submissions/DmytryjK/js-oop-exercise-tiny/index.js

```diff
@@ -0,0 +1,92 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+class Inhabitant {
+   constructor(name, gender, saying) {
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+   }
+   print() {
+      const specs = [this.species, this.name, this.gender, this.legs, this.hands, this.saying];
+      print(specs.join("; "));
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(name, gender, saying) {
+      super(name, gender, saying);
+      this.legs = 4; 
+      this.hands = 0;
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying) {
+      super(name, gender, saying);
+      this.species = 'human';
```

Comment:
`species` is a shared property of all inhabitants. It should be defined at some base class. This helps to keep code DRY and have any related in a single place.